### PR TITLE
Homework 2

### PR DESCRIPTION
**Pull request by kali-03**

### CODE REVIEW

File: module2/homework/nwdNww/nwdNww.hpp [hunk 1]
```diff
@@ -1,11 +1,41 @@
...
+    }
+
+    if ((lhs<0 && rhs<0) || rhs<0){
+        return (rhs*(-1));
+    }
+
+    if (lhs<0 && rhs>0){
+        return rhs;
+    }
+
```
Comment (piezol) [inline module2/homework/nwdNww/nwdNww.hpp:20]: You've made 4 'edge cases' that work differently than generic case.
The more edge cases there are, the harder the code is to analyze by someone who reads it in the future.
Try to think about the ways to minimalize the number of 'if-then-returns'.

File: module2/homework/nwdNww/nwdNww.hpp [hunk 2]
```diff
@@ -1,11 +1,41 @@
...
+    if (lhs==0 || rhs==0){
+        return 0;
+    }
+
+    if (lhs==1 || rhs==1){
+        return 1;
+    }
+    else{
+        return (lhs*rhs)/NWD(lhs,rhs);
+    }
```
Comment (DamiXioR) [inline module2/homework/nwdNww/nwdNww.hpp:40]: i think you can skip last else and use just return

File: module2/homework/maxOfVector/maxOfVector.hpp [hunk 3]
```diff
@@ -1,8 +1,15 @@
...
+#include <algorithm>
 
 int maxOfVector(const std::vector<int>& vec) {
-    // TODO: Implement me :)
-    return {};
+    int max=vec[0];
+     
+    for (const auto& el:vec){
+        if (el>vec[0])
+          max =el;
```
Comment (DamiXioR) [inline module2/homework/maxOfVector/maxOfVector.hpp:11]: in my opinion you should use curly braces even with oneline if statement

File: module2/homework/maxOfVector/maxOfVector.hpp [hunk 4]
```diff
@@ -1,8 +1,15 @@
...
 #include <limits>
+#include <algorithm>
 
 int maxOfVector(const std::vector<int>& vec) {
-    // TODO: Implement me :)
-    return {};
+    int max=vec[0];
+     
+    for (const auto& el:vec){
+        if (el>vec[0])
```
Comment (DamiXioR) [inline module2/homework/maxOfVector/maxOfVector.hpp:10]: i think the better is verify el with max variable

File: module2/homework/addEven/addEven.hpp [hunk 5]
```diff
@@ -2,7 +2,10 @@
 #include <vector>
 
 int addEven(const std::vector<int>& numbers) {
-    // TODO: Your implementation goes here
-    // Below return is only to make this function compile now
-    return -1;
+    int result=0;
+    for (const auto&el:numbers){
```
Comment (nauka-programowania-MA) [inline module2/homework/addEven/addEven.hpp:6]: space

File: module2/homework/addEven/addEven.hpp [hunk 6]
```diff
@@ -2,7 +2,10 @@
...
+        if (el%2==0)
+           result+=el;
+     }
+    return result;
```
Comment (nauka-programowania-MA) [inline module2/homework/addEven/addEven.hpp:10]: space between operator like & or += etc... and in for loops

File: module2/homework/generateSequence/generateSequence.hpp [hunk 7]
```diff
@@ -2,6 +2,14 @@
...
+    if (count<0){
+      return {};
+    }
+
+    std::vector<int> vec(count);
+    for (int i=0; i < count; i++){
+        vec[i]=(i+1)*step; 
+    }
+    return {vec};
 }
```
Comment (nauka-programowania-MA) [inline module2/homework/generateSequence/generateSequence.hpp:15]: format

File: module2/homework/maxOfVector/maxOfVector.hpp [hunk 8]
```diff
@@ -1,8 +1,15 @@
 #pragma once
 #include <vector>
 #include <limits>
+#include <algorithm>
```
Comment (nauka-programowania-MA) [inline module2/homework/maxOfVector/maxOfVector.hpp:4]: please reorder includes alphabetically

File: module2/homework/maxOfVector/maxOfVector.hpp [hunk 9]
```diff
@@ -1,8 +1,15 @@
...
-    // TODO: Implement me :)
-    return {};
+    int max=vec[0];
+     
+    for (const auto& el:vec){
+        if (el>vec[0])
+          max =el;
+    }
+
+    return max;
```
Comment (nauka-programowania-MA) [inline module2/homework/maxOfVector/maxOfVector.hpp:14]: format

File: module2/homework/maxOfVector/maxOfVector.hpp [hunk 10]
```diff
@@ -1,8 +1,15 @@
 #pragma once
 #include <vector>
 #include <limits>
```
Comment (nauka-programowania-MA) [inline module2/homework/maxOfVector/maxOfVector.hpp:3]: remove if not used

File: module2/homework/nwdNww/nwdNww.hpp [hunk 11]
```diff
@@ -1,11 +1,41 @@
...
+        return lhs;
+    }
+
+    if ((lhs<0 && rhs<0) || rhs<0){
+        return (rhs*(-1));
+    }
+
+    if (lhs<0 && rhs>0){
+        return rhs;
+    }
```
Comment (nauka-programowania-MA) [inline module2/homework/nwdNww/nwdNww.hpp:19]: format

Comment (nauka-programowania-MA) [inline module2/homework/nwdNww/nwdNww.hpp:19]: try use if else

File: module2/homework/nwdNww/nwdNww.hpp (hunk 2 above)
Comment (nauka-programowania-MA) [inline module2/homework/nwdNww/nwdNww.hpp:40]: format

**Comment by ziobron** [comment]
+4 XP addEven
+6 XP NwwNwd
+5 XP maxOfVector
+5 XP generateSequence
+8 XP delivery before 31.05

{ "requirement_ids": [1119, 1631, 1119, 1633, 1119, 1634, 1119, 1635], "points": [2, 4, 2, 6, 2, 5, 2, 5] }

**Pull request closed without merging**